kvbc/tools/object_store_utility/integrity_checker.cpp

```diff
@@ -216,32 +228,49 @@ void IntegrityChecker::validateKey(const std::string& key) const {
   // retrieve block and get value
   auto [block_digest, raw_block] = getBlock(containing_block_id);
   (void)block_digest;
-  for (auto& [category, value] : raw_block.data.updates.kv) {
-    auto cat = category;
-    std::visit(
-        [cat, key, this](auto&& arg) {
-          using T = std::decay_t<decltype(arg)>;
-          if constexpr (std::is_same_v<T, kvbc::categorization::BlockMerkleInput>) {
-            auto search = arg.kv.find(key);
-            if (search == arg.kv.end())
-              LOG_DEBUG(logger_, "key [" << key << "] not found in category: " << cat);
-            else
-              LOG_INFO(logger_, "found key [ " << key << "] value [" << search->second << "] in category " << cat);
-          } else if constexpr (std::is_same_v<T, kvbc::categorization::VersionedInput>) {
-            auto search = arg.kv.find(key);
-            if (search == arg.kv.end())
-              LOG_DEBUG(logger_, "key [" << key << "] not found in category: " << cat);
-            else
-              LOG_INFO(logger_, "found key [ " << key << "] value [" << search->second.data << "] in category " << cat);
-          } else if constexpr (std::is_same_v<T, kvbc::categorization::ImmutableInput>) {
-            auto search = arg.kv.find(key);
-            if (search == arg.kv.end())
-              LOG_DEBUG(logger_, "key [" << key << "] not found in category: " << cat);
-            else
-              LOG_INFO(logger_, "found key [ " << key << "] value [" << search->second.data << "] in category " << cat);
-          }
-        },
-        value);
+  std::optional<concord::kvbc::categorization::CategoryInput> category_input;
```

Comment:
I think that each block must have the category_input, maybe it's worth asserting that category_input.has_vaue()? wdyt?

Reply:
Thanks, @cloudnoize for the comment. I think this should be checked. I saw that in Line 244 this check is present.